### PR TITLE
[Testerina] Single code coverage report XML improvements

### LINKED ISSUE #29677: [Testerina] Sourcefile and class coverages are duplicated for multi-module projects in covergae-report.xml
**Description:**
$subject. This needs to be fixed to merge duplicated coverages preserving the coverage information.

Anyway, the line coverage information is processed and shown correctly in tools like codecov. But complexity information shown for ballerina files for multi module projects is incorrect.

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Dilhasha**
Fixes #29677 #29653

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package org.ballerinalang.test.runtime.entity;
```
Comment (NipunaRanasinghe) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java:1]: Missing license header

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Override
+    public boolean containsCode() {
+        return oldMethodCoverage.containsCode();
+    }
+
+    @Override
+    public ICoverageNode getPlainCopy() {
+        return oldMethodCoverage.getPlainCopy();
+    }
```
Comment (NipunaRanasinghe) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java:104]: `oldMethodCoverage` field is accessed using `this` instance in some getters and without `this` in the others. Shall we make the usages consistent?  WDYT?

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java (hunk 1 above)
Comment (Dilhasha) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java:18]: Fixed with dbb286f7c536e9c5ebec4378ece5fa75e8166b34

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java (hunk 2 above)
Comment (Dilhasha) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/NormalizedCoverageMethod.java]: Fixed with dbb286f7c536e9c5ebec4378ece5fa75e8166b34

**Review by NipunaRanasinghe — COMMENTED** [review]
LGTM

**Review by azinneera — APPROVED** [review]

**Pull request merged**